plan/aggregation_push_down.go

```diff
@@ -243,53 +243,56 @@ func (a *aggregationOptimizer) getDefaultValues(agg *LogicalAggregation) ([]type
 	return defaultValues, true
 }
 
-func (a *aggregationOptimizer) checkAnyCountAndSum(aggFuncs []aggregation.Aggregation) bool {
+func (a *aggregationOptimizer) checkAnyCountAndSum(aggFuncs []*aggregation.AggFuncDesc) bool {
 	for _, fun := range aggFuncs {
-		if fun.GetName() == ast.AggFuncSum || fun.GetName() == ast.AggFuncCount {
+		if fun.Name == ast.AggFuncSum || fun.Name == ast.AggFuncCount {
 			return true
 		}
 	}
 	return false
 }
 
-func (a *aggregationOptimizer) makeNewAgg(aggFuncs []aggregation.Aggregation, gbyCols []*expression.Column) *LogicalAggregation {
+func (a *aggregationOptimizer) makeNewAgg(aggFuncs []*aggregation.AggFuncDesc, gbyCols []*expression.Column) *LogicalAggregation {
 	agg := LogicalAggregation{
 		GroupByItems: expression.Column2Exprs(gbyCols),
 		groupByCols:  gbyCols,
 	}.init(a.ctx)
 	aggLen := len(aggFuncs) + len(gbyCols)
-	newAggFuncs := make([]aggregation.Aggregation, 0, aggLen)
+	newAggFuncDescs := make([]*aggregation.AggFuncDesc, 0, aggLen)
 	schema := expression.NewSchema(make([]*expression.Column, 0, aggLen)...)
 	for _, aggFunc := range aggFuncs {
-		var newFuncs []aggregation.Aggregation
+		var newFuncs []*aggregation.AggFuncDesc
 		newFuncs, schema = a.decompose(aggFunc, schema, agg.ID())
-		newAggFuncs = append(newAggFuncs, newFuncs...)
+		newAggFuncDescs = append(newAggFuncDescs, newFuncs...)
 	}
 	for _, gbyCol := range gbyCols {
-		firstRow := aggregation.NewAggFunction(ast.AggFuncFirstRow, []expression.Expression{gbyCol.Clone()}, false)
-		newAggFuncs = append(newAggFuncs, firstRow)
+		firstRow := aggregation.NewAggFuncDesc(ast.AggFuncFirstRow, []expression.Expression{gbyCol.Clone()}, false)
+		firstRow.TypeInfer(agg.ctx)
```

Comment:
How about just combine the `TypeInfer` and `NewFuncDesc` for the complete model? Almost every `NewFuncDesc` is followed by `TypeInfer`.